pyproject.toml

```diff
@@ -28,16 +28,20 @@ types-pyyaml = "^6.0.12.12"
 jsonschema = "^4.2.0"
 types-jsonschema = "^4.2.0"
 prompt-toolkit = "^3.0.39"
-pinecone-text = "^0.7.2"
+pinecone-text = [{version = "^0.7.2"},
+                 {version = "^0.7.2", extras = ["dense"], optional = true}]
+
 tokenizers = "^0.15.0"
-transformers = "^4.35.2"
+transformers = {version = "^4.35.2", optional = true}
 sentencepiece = "^0.1.99"
 pandas = "2.0.0"
 pyarrow = "^14.0.1"
 cohere = { version = ">=4.37", optional = true }
 
 [tool.poetry.extras]
 cohere = ["cohere"]
+torch = ["pinecone-text"]
```

Comment:
Are you sure this could be `torch` and not have to be `dense`? from the link you sent the used the extras name but maybe it's just in the example and not a must

Reply:
It can be anything you want, it's just a name (I tested it).  
When you `poetry install -E torch` it's interpreted as "the `optional` version of `pinecone-text`, as defined above".